app/controllers/steps_controller.rb

```diff
@@ -0,0 +1,27 @@
+class StepsController < ApplicationController
+  include Wicked::Wizard::Translated
+
+  before_action :authenticate_user!
+
+  steps(*User.steps)
+
+  def show
+    render_wizard
+  end
+
+  def update
+    wizard_value = wizard_value(step)
+
+    if wizard_value != "stage_5"
+      if wizard_value >= current_user.highest_aasm_state
+        current_user.complete_stage
+      end
+
+      wizard_path = current_user.highest_aasm_state.gsub("stage_", "")
```

Comment:
Hack to "reverse-reverse translate" the furthest page the User has visited so we can set the internal state of Wicked to redirect to that furthest page when we call `render_wizard current_user`.